### PR TITLE
feat(scheduler): cap concurrent workflows per compute backend

### PR DESCRIPTION
**Pull request by CameronMcClymont**
Workflows whose steps run on external backends (HTCondor, Slurm) consume fewer Kubernetes resources, but the scheduler does not take this into account. Even fully external workflows are still subject to the concurrency limit.

This PR introduces more generous concurrency limits for external workflows rather than subjecting them to a coarse global limit.

Closes https://github.com/reanahub/reana-workflow-controller/issues/663

### CODE REVIEW

File: reana_server/complexity.py [hunk 1]
```diff
@@ -41,14 +41,20 @@ def validate_job_memory_limits(complexity: List[Tuple[int, float]]) -> None:
...
+    :param complexity: list of (k8s_jobs, memory_bytes) tuples, one per initial
+        workflow step. k8s_jobs is 0 for steps that run on external backends
+        (HTCondor, Slurm) and >0 for steps that run on Kubernetes.
+    :return: minimum memory in bytes across initial Kubernetes steps, or 0 if
+        there are no initial Kubernetes steps (e.g. the workflow starts on an
+        external backend). A return value of 0 causes the Kubernetes node memory
+        check to be skipped entirely.
     """
-    if not complexity:
+    k8s_memories = [mem for jobs, mem in complexity if jobs > 0]
```
Comment (michaelbuchar) [inline reana_server/complexity.py:55]: This fixes estimators that already mark external-backend steps with `jobs=0`, but Snakemake does not seem to do that today. Since Snakemake supports per-step `compute_backend`, external-backend Snakemake workflows may still end up with a non-zero `min_job_memory` and still hit the Kubernetes memory check. Should we handle Snakemake here as well, or treat this as a partial fix?

Comment (CameronMcClymont) [reply reana_server/complexity.py:55]: Fixed to skip Kubernetes memory check for external Snakemake steps in line with other workflow languages.

File: reana_server/scheduler.py [hunk 2]
```diff
@@ -71,7 +80,23 @@ def check_concurrent_workflows_limit() -> Optional[str]:
...
+                    func.array_length(Workflow.complexity, 1).is_(None),
+                    # Count only workflows that have at least one Kubernetes step.
+                    # ``complexity`` is a 2-D array of [job_count, memory] rows
+                    # (Postgres arrays are 1-indexed), so ``complexity[i][1]`` is
+                    # the job count of step ``i`` and is >0 for Kubernetes steps.
+                    text(
+                        "EXISTS ("
+                        "SELECT 1 FROM generate_series(1, array_length(complexity, 1)) AS i "
+                        "WHERE complexity[i][1] > 0"
+                        ")"
```
Comment (michaelbuchar) [inline reana_server/scheduler.py]: Hybrid workflows are still dropped from the running-workflow count after submission, so #780 can under-enforce the Kubernetes concurrency cap. The PR correctly adds all-step classification via workflow_uses_kubernetes() in reana_server/complexity.py:83-96, and publish_workflow_submission() even notes that stored complexity only reflects initial steps in reana_server/utils.py:399-413. But check_concurrent_workflows_limit() still counts already-running workflows only from Workflow.complexity in reana_server/scheduler.py:64-91. For a workflow that starts on HTCondor/Slurm and later runs on Kubernetes, those stored complexity rows can still be all jobs=0, so the workflow will not consume a concurrency slot once it is queued/running. I reproduced the underlying mismatch locally with a minimal serial hybrid spec: estimate_complexity() returned [(0, 2147483648.0)] for an external-first, later-Kubernetes workflow. The new scheduler tests in tests/test_scheduler.py:190-224 cover only K8S_COMPLEXITY vs EXTERNAL_COMPLEXITY; they do not cover the critical case uses_kubernetes=True with initial-step complexity=[(0, ...)].

Comment (CameronMcClymont) [reply reana_server/scheduler.py]: I have pushed a fix for this but haven't squashed it because a cleaner option would be to add a `uses_kubernetes` column in reana-db which would make the code in this PR a lot cleaner with less 'magic values' and commenting necessary. I will open the reana-db PR and push one more commit here and then we can choose which route to go down.

Comment (CameronMcClymont) [reply reana_server/scheduler.py]: Tbh, the reana-db route is much much cleaner and avoids forcing this 'does this workflow ever touch Kubernetes' boolean through the `complexity` column when it should really have its own. reana-db PR is here: https://github.com/reanahub/reana-db/pull/267

**Review by michaelbuchar — DISMISSED** [review]
This looks good to me, only thing missing is fixing the CI checks!

File: reana_server/scheduler.py [hunk 3]
```diff
@@ -61,8 +61,15 @@ def check_memory_availability(
...
-def check_concurrent_workflows_limit() -> Optional[str]:
-    """Check upper limit on running REANA batch workflows."""
+def check_concurrent_workflows_limit(uses_kubernetes: bool = True) -> Optional[str]:
+    """Check upper limit on running REANA batch workflows.
+
+    Workflows that run exclusively on external backends (HTCondor, Slurm) do not
+    consume Kubernetes resources: they bypass this check and are excluded when
+    counting already-running workflows.
+    """
+    if not uses_kubernetes:
```
Comment (tiborsimko) [inline reana_server/scheduler.py]: **🤖 RWC663-02: `REANA_MAX_CONCURRENT_BATCH_WORKFLOWS` is silently neutered for external workflows (HIGH)**

The PR turns the cluster-wide concurrency cap into an advisory cap that only applies to Kubernetes-job workflows. External-only workflows neither check the cap on their own admission nor count toward it for any future check.

`reana-server/reana_server/scheduler.py:64-86`:

```python
def check_concurrent_workflows_limit(uses_kubernetes: bool = True) -> Optional[str]:
    if not uses_kubernetes:
        return None                       # (a) external workflow → no cap check
    error = None
    try:
        running_workflows = (
            Session.query(func.count())
            .filter(or_(Workflow.status == RunStatus.pending,
                        Workflow.status == RunStatus.running),
                    Workflow.uses_kubernetes) # (b) external workflows don't count
            .scalar()
        )
        if running_workflows >= REANA_MAX_CONCURRENT_BATCH_WORKFLOWS:
            ...
```

A user submitting 10,000 HTCondor-only workflows admits 10,000 workflow-engine pods on Kubernetes. The engine always runs on k8s — it is the orchestrator pod that talks to HTCondor/Slurm — and lives for the entire wall-clock duration of the external jobs, which for HTCondor can be hours or days. The original cap was, among other things, protection against exactly this kind of orchestrator-pod explosion, and that protection is silently gone after upgrade. `REANA_MAX_CONCURRENT_BATCH_WORKFLOWS` is the only global workflow-count cap REANA has — there is no per-backend cap, no per-user concurrency cap, and quotas are tracked post-consumption rather than as admission control. (Per-workflow Dask caps exist — `REANA_DASK_CLUSTER_MAX_MEMORY_LIMIT`, `REANA_DASK_CLUSTER_MAX_NUMBER_OF_WORKERS` — but nothing caps the *number of concurrent Dask workflows*.)

Two further pieces of evidence the cap's semantics shifted in ways the PR does not acknowledge:

1. **Dashboard and scheduler disagree.** `ClusterHealth.get_workflow_health()` (`reana-server/reana_server/status.py:655-676`) reports `running + pending` against `REANA_MAX_CONCURRENT_BATCH_WORKFLOWS` and **does not** filter by `uses_kubernetes`. After this PR the dashboard will display "30/30, cluster full" while the scheduler is still admitting workflows. Operators reading the dashboard will be misled.
2. **Per-user priority and scheduler cap disagree.** `User.get_workflow_overload_priority()` (`reana-db/reana_db/models.py:329-351`) counts pending+running **without** filtering by `uses_kubernetes`, so external workflows still depress the user's priority. Two views of the same population, different filters.

Suggested fix — preserve the cap as a safety guarantee rather than removing it for external workflows. The right shape is **per-backend caps** along the compute-backend axis, optionally complemented by a **per-resource cap** along the Dask axis. Sketch:

```
REANA_MAX_CONCURRENT_K8S_BATCH_WORKFLOWS       = 30   # renamed-with-meaning of the
                                                       # current variable; counts
                                                       # workflows using the k8s backend
REANA_MAX_CONCURRENT_HTCONDOR_BATCH_WORKFLOWS  = 200  # new; bounds engine pods for
                                                       # HTCondor-using workflows
REANA_MAX_CONCURRENT_SLURM_BATCH_WORKFLOWS     = 200  # new; same for Slurm

REANA_MAX_CONCURRENT_DASK_WORKFLOWS            = 5    # new; orthogonal to the above,
                                                       # bounds workflows that request
                                                       # a Dask cluster regardless of
                                                       # their step backends
```

The defaults are illustrative, not prescriptive — operators pick numbers that match their cluster. Each cap is applied independently in `check_concurrent_workflows_limit()`; a hybrid k8s+HTCondor workflow counts against both backend caps, a Dask-on-HTCondor workflow counts against the HTCondor cap and the Dask cap. This composes naturally with the data-model change in [finding RWC663-03](https://github.com/reanahub/reana-db/pull/267#discussion_r3421389305) (`Workflow.compute_backends` ARRAY for the backend axis, the existing `workflow.services` relation for Dask).

Important: the per-DAG memory awareness must stay. `check_memory_availability()` already computes `workflow_min_job_memory` from the spec at submission time (`reana-server/reana_server/utils.py:417` → `complexity.py:get_workflow_min_job_memory`) and checks it against the largest node's free memory before admission. This is exactly the "is a DAG's smallest k8s step able to land on the cluster right now?" check that effective scheduling depends on, and the PR correctly keeps it. The cap rework should add per-backend slot accounting **on top of** the existing memory check, not replace it.

Whichever shape lands, three things must move together:

- `ClusterHealth.get_workflow_health()` updated so the dashboard reports against the same per-backend counts the scheduler uses (one tile per backend, or an aggregate that matches the scheduler's admission decision exactly).
- `User.get_workflow_overload_priority()` aligned so the priority calculation and the cap calculation see the same population.
- `REANA_WORKFLOW_SCHEDULING_READINESS_CHECK_LEVEL` documentation refreshed: the existing `concurrent` / `memory` / `all_checks` knob (`reana-server/reana_server/config.py:208-230`) currently implies a single concurrency check; it now needs to call out that "concurrent" is the OR of all per-backend caps.

A weighted-cap variant — count an external workflow as e.g. 0.1 of a slot, sum-based admission — would also preserve the original safety property and is mechanically simpler than multiple caps, but at the cost of operator legibility: the cap stops meaning "this many workflows" and starts meaning "this much weighted footprint", which is harder to set and reason about than a per-backend number that maps onto operator intuitions about HTCondor capacity, Dask capacity, etc.

File: reana_server/complexity.py [hunk 4]
```diff
@@ -57,29 +77,28 @@ def estimate_complexity(workflow_type, reana_yaml):
...
-    estimator = build_estimator(workflow_type, reana_yaml)
+    estimator = _build_estimator(workflow_type, reana_yaml)
     try:
         complexity = estimator.estimate_complexity()
     except Exception:
         return []
     return complexity
 
 
+def workflow_uses_kubernetes(workflow_type: str, reana_yaml: dict) -> bool:
```
Comment (tiborsimko) [inline reana_server/complexity.py]: **🤖 RWC663-01: Dask workflows with external step backends bypass the concurrent-workflow cap despite creating heavy Kubernetes resources (HIGH)**

A note on scope before the framing: this finding is **not** about the standard Dask demo. In a typical Dask workflow the steps omit `compute_backend` and therefore default to Kubernetes; the classifier sees k8s steps, marks the workflow `uses_kubernetes=True`, and the cap applies normally. The bypass triggers only when both conditions hold: `workflow.resources.dask` is set, **and** every step explicitly uses an external backend (HTCondor or Slurm). That combination is rarer than a vanilla Dask demo but it is valid configuration, and the workflows it admits are precisely the heaviest ones.

The framing that matters: a Dask-enabled workflow is not a "lighter" workflow that happens to escape a cap on a technicality. It is among the **heaviest** Kubernetes consumers REANA admits. Per-workflow defaults (`reana-server/reana_server/config.py:112-134` and the Helm chart) cap the requested Dask cluster at `REANA_DASK_CLUSTER_MAX_MEMORY_LIMIT = 16Gi` and `REANA_DASK_CLUSTER_MAX_NUMBER_OF_WORKERS = 20`. A perfectly honest user specifying `number_of_workers: 20, single_worker_memory: 800Mi` lands on 16 GiB of long-lived worker memory plus a scheduler pod plus the workflow-engine pod — all on Kubernetes, all for the wall-clock duration of the workflow. This is a much larger footprint than a typical batch workflow whose initial step requests a few GiB.

The cluster-wide `REANA_MAX_CONCURRENT_BATCH_WORKFLOWS` cap exists precisely to put a ceiling on this kind of resource commitment. After this PR, the specific class of Dask workflows described above is the one that escapes it. Direction of the bug matters: the cap stops applying along the axis where it is most needed.

The reason is a scope mistake in the classifier. `workflow_uses_kubernetes()` only inspects step-level `compute_backend`. Every estimator (`SerialComplexityEstimator`, `YadageComplexityEstimator`, `CWLComplexityEstimator`, `SnakemakeComplexityEstimator`) walks the workflow step tree and asks the same question — "is the step's compute backend `kubernetes`?" — and nothing else.

`reana-server/reana_server/complexity.py:88-99`:

```python
def workflow_uses_kubernetes(workflow_type: str, reana_yaml: dict) -> bool:
    """Return True if any step across the entire workflow runs on Kubernetes.
    ...
    """
    try:
        return _build_estimator(workflow_type, reana_yaml).uses_kubernetes()
    except Exception:
        return True
```

But Dask is declared at workflow-resource level, completely outside the step tree — `reana-server/reana_server/utils.py:813-817`:

```python
def workflow_uses_dask(reana_specification: Dict) -> bool:
    """Check whether a workflow specification requests a Dask service."""
    return bool(
        reana_specification.get("workflow", {}).get("resources", {}).get("dask")
    )
```

`DaskResourceManager` (`reana-workflow-controller/reana_workflow_controller/dask.py:51`, instantiated from `workflow_run_manager.py:407-410`) creates a `DaskCluster` CR with a scheduler pod plus N worker pods, optionally with an autoscaler.

So a workflow whose steps run on HTCondor or Slurm but whose `workflow.resources.dask` requests the maximum-allowed Dask cluster — e.g.:

```yaml
workflow:
  type: serial
  resources:
    dask:
      image: ...
      number_of_workers: 20
      single_worker_memory: 800Mi   # 20 × 800Mi = 16 GiB, the cluster max
  specification:
    steps:
      - compute_backend: htcondor
        commands: [...]
```

…classifies as external → the scheduler returns early from `check_concurrent_workflows_limit` (`reana-server/reana_server/scheduler.py:71`) → cap bypassed → ~16 GiB of Dask cluster memory plus a scheduler pod plus the engine pod spawn on the REANA cluster, unmetered, for each such workflow admitted. The cap that would otherwise limit how many of these accumulate at once is silent. The user does not need to "pretend" anything — straightforward use of documented Dask + documented external-backend features triggers this.

Suggested fix — short-circuit the classifier on any workflow-level resource that materialises as a Kubernetes object. Minimal patch:

```python
def workflow_uses_kubernetes(workflow_type: str, reana_yaml: dict) -> bool:
    # Workflow-level resources that materialise as k8s objects:
    if reana_yaml.get("workflow", {}).get("resources", {}).get("dask"):
        return True
    try:
        return _build_estimator(workflow_type, reana_yaml).uses_kubernetes()
    except Exception:
        return True
```

This is the narrowest patch that unblocks merge. Note that it still treats a Dask workflow as a single "k8s slot" even though it has a much larger k8s footprint than a regular workflow. A more honest accounting — a weighted cap, or a per-workflow k8s-resource profile that the scheduler can sum against the cap — is the natural follow-up and is the subject of separate findings on [the cap's semantics](https://github.com/reanahub/reana-server/pull/780#discussion_r3421401195) and on [the persisted data model](https://github.com/reanahub/reana-db/pull/267#discussion_r3421389305).

File: reana_server/scheduler.py (hunk 3 above)
Comment (CameronMcClymont) [reply reana_server/scheduler.py]: Addressed by introducing per-backend concurrency limits rather than the weighted-cap variant.

File: reana_server/complexity.py [hunk 5]
```diff
@@ -286,6 +352,14 @@ def parse_specification(self, initial_step):
 class CWLComplexityEstimator(ComplexityEstimatorBase):
     """REANA CWL workflow complexity estimation."""
 
+    @staticmethod
+    def _get_step_hints(step):
```
Comment (tiborsimko) [inline reana_server/complexity.py:356]: **🤖 RWC663-04: Backend classification can disagree with what actually executes (CWL and Yadage) (MEDIUM)**

The new per-backend caps are only as good as the classifier that assigns workflows to backends, and for CWL and Yadage the classifier extracts a step's `compute_backend` with different semantics than both the validator and the workflow engines.

`reana_server/complexity.py:356`:

```python
@staticmethod
def _get_step_hints(step):
    """Return the last hint of a CWL step, or an empty dict if none.
    ...
    """
    return (step.get("hints") or [{}])[-1]
```

Four code paths, three distinct semantics for CWL:

- **Classifier (this PR)**: last hint only, `hints[-1]`.
- **Validator**: scans all hints for `class == "reana"` — reana-commons' `ComputeBackendValidatorCWL._get_reana_hints`.
- **Engine**: scans all hints, first hint containing the key wins — reana-workflow-engine-cwl's `_get_hint` in `cwl_reana.py`.
- The complexity estimator's job counting shares the classifier's last-hint assumption (as it did before this PR via `step.get("hints", [{}]).pop()`).

The canonical REANA representation is a `class: reana` hint, and nothing guarantees it is last. A step declared as:

```yaml
hints:
  - class: reana
    compute_backend: htcondorcern
  - class: ResourceRequirement
    ramMin: 4096
```

validates and executes as HTCondor, but the classifier reads `compute_backend` off the `ResourceRequirement` dict, gets nothing, and classifies the step as Kubernetes. The persisted `Workflow.compute_backends` is then wrong: the workflow is charged against the Kubernetes cap it does not use, and the HTCondor cap undercounts.

Yadage has the mirror-image problem: the classifier (`_get_stage_compute_backend`, `reana_server/complexity.py:195`) and the reana-commons validator both take the *first* `compute_backend` resource, while the engine's `_get_resources` in reana-workflow-engine-yadage's `externalbackend.py` iterates all resources with overwrite, so the *last* one wins. A stage declaring:

```yaml
resources:
  - compute_backend: htcondorcern
  - compute_backend: kubernetes
```

passes validation (first entry), is classified as external-only, and therefore bypasses both the Kubernetes concurrency cap and the Kubernetes memory check — yet the engine runs the job on Kubernetes. Nothing rejects duplicate declarations.

The new tests do not catch any of this because they exercise only the classifier, using a synthetic single-hint shape without `class: reana` (`hints: [{"compute_backend": "htcondor"}]`). On that shape the classifier and the engine happen to agree, while the validator's canonical REANA-hint semantics are never exercised.

On scope, to be fair: the CWL last-hint assumption predates this PR, so this is a pre-existing weakness rather than a regression — but this PR makes it load-bearing by persisting the classification and hanging admission control off it, so it is best fixed here.

Suggested fix — centralise "effective backend of a step" extraction so it matches the execution semantics: for CWL, scan all hints for the `class: reana` hint (matching the validator; the engine's first-key-wins rule is itself subtly different and worth reconciling in the same pass); for Yadage, either align on the engine's last-wins behaviour or, better, reject multiple `compute_backend` declarations per step at validation time in reana-commons. Add a realistic multi-hint CWL test (a `class: reana` hint followed by another hint) and a duplicate-resource Yadage test.

File: reana_server/status.py [hunk 6]
```diff
@@ -653,25 +656,61 @@ def get_job_health(self):
...
+        binding = backends[bottleneck]
         return {
             "running": running,
             "queued": queued,
             "pending": pending,
-            "available": available,
-            "total": total,
-            "percentage": percentage,
-            "health": ClusterHealth.get_health_status(percentage),
+            "backends": backends,
```
Comment (tiborsimko) [inline reana_server/status.py:707]: **🤖 RWC663-07: `/api/status` response grew fields the API contract does not describe (MEDIUM)**

`get_workflow_health()` now returns `backends`, `bottleneck` and `used` alongside the old keys (`reana_server/status.py:707-712`), and reana-ui PR #516 consumes them. But the OpenAPI docstring on the `/api/status` endpoint (`reana_server/rest/status.py`) still describes only the old single-counter `workflow` object (`available`/`running`/`queued`/`pending`/`percentage`/`health`/`sort`/`total`), and its `examples:` block still shows the old payload. Because `docs/openapi.json` is generated from that same docstring, the `--docs-openapi` check cannot detect the omission — the generated and committed specifications are both stale and still match — and the bundled copy in `reana-commons/reana_commons/openapi_specifications/reana_server.json`, the contract that bravado clients load at runtime, is equally stale. The headline feature of this PR set is thus invisible in the very API contract the new UI depends on.

Relatedly, nothing exercises the new response shape: `test_status.py` was not extended, so the per-backend tile assembly and the bottleneck selection — the only substantive new logic in `status.py` — have no test coverage.

Suggested fix — extend the docstring schema with the new `workflow` properties (`backends` as an object of per-resource `{available, health, percentage, total, used}` entries, plus `bottleneck` and `used`) and update the example payload to match, remembering that apispec emits schema fields in alphabetical order; regenerate `docs/openapi.json` via `./run-tests.sh --docs-openapi` and copy the result to `reana-commons/reana_commons/openapi_specifications/reana_server.json`. Add a `get_workflow_health()` unit test covering bottleneck selection (e.g. a saturated external backend against a roomy Kubernetes cap).

File: reana_server/status.py [hunk 7]
```diff
@@ -653,25 +656,61 @@ def get_job_health(self):
...
 
+        active_counts = Workflow.count_active_per_backend()
+
+        # One tile per configured compute backend (so operators see headroom even
+        # at zero usage), plus any backend currently in use, plus the Dask cap.
+        resources = (
+            set(SUPPORTED_COMPUTE_BACKENDS) | {"kubernetes"} | set(active_counts)
+        )
+        resources.discard("dask")
+        resources = sorted(resources) + ["dask"]
```
Comment (tiborsimko) [inline reana_server/status.py:682]: **🤖 RWC663-10: Dask capacity is reported even when Dask is disabled (LOW)**

`get_workflow_health()` unconditionally appends `"dask"` to the reported resources (`reana_server/status.py:681-682`):

```python
resources.discard("dask")
resources = sorted(resources) + ["dask"]
```

On a deployment with `DASK_ENABLED=false`, workflow specifications requesting Dask are rejected at validation (`reana_server/validation.py:192`), so no new Dask workflows can be admitted — yet the Workflows tile's backend breakdown and tooltips still show a Dask entry with, e.g., `0/10` slots. In steady state its count is 0, so it can never become the bottleneck, and the impact is confined to the breakdown misleading operators into thinking Dask capacity exists on a cluster where it does not. (The count is not *guaranteed* to be 0: if Dask is disabled while a Dask workflow is still active, that workflow legitimately shows up in the count until it finishes.)

Suggested fix — show the entry only when Dask is enabled or transitional usage is actually present, so nonexistent idle capacity is hidden without hiding real usage:

```python
from reana_server.config import DASK_ENABLED
...
resources.discard("dask")
resources = sorted(resources)
if DASK_ENABLED or active_counts.get("dask", 0):
    resources.append("dask")
```

reana-ui needs no matching change: it renders whatever `backends` contains.

File: reana_server/complexity.py (hunk 4 above)
Comment (tiborsimko) [reply reana_server/complexity.py]: The amended code resolves this finding: `publish_workflow_submission` now publishes `uses_dask` alongside the submission message, the scheduler applies the new orthogonal `REANA_MAX_CONCURRENT_DASK_WORKFLOWS` cap, and active Dask workflows are counted via the `Workflow.services` relation. `test_dask_on_external_workflow_submission_is_flagged` and `test_dask_workflows_are_capped` cover exactly the Dask-on-external scenario described here. Considering this addressed — thanks!

File: reana_server/complexity.py (hunk 5 above)
Comment (CameronMcClymont) [reply reana_server/complexity.py:356]: Very nice catch. Fixed now so both classifiers now resolve a step's backend the way the engine that runs it does, so the persisted `compute_backends` matches what actually executes. Also added new tests to cover the multi-hint CWL case and the duplicate-resource Yadage case in both orders, plus `tests/test_compute_backends.py` in reana-commons for the validator side.

File: reana_server/status.py (hunk 6 above)
Comment (CameronMcClymont) [reply reana_server/status.py:714]: Updated so the API contract now describes the response the UI actually consumes. `docs/openapi.json` was regenerated too and copied into `reana_commons/openapi_specifications/reana_server.json` in reanahub/reana-commons#541.

File: reana_server/status.py (hunk 7 above)
Comment (CameronMcClymont) [reply reana_server/status.py]: Fixed, using your suggested snippet.
